onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlReservedResourceSubAllocator.cpp

```diff
@@ -0,0 +1,385 @@
+            for (auto& entry : m_outstandingAllocationsById)
+            {
+                printf("%u\n", static_cast<int>(entry.first));
+            }
+            printf("\n");
+        }
+#endif
+    }
+
+    /*static*/ gsl::index DmlReservedResourceSubAllocator::GetBucketIndexFromSize(uint64_t size)
+    {
+        assert(size != 0);
+
+        // Each bucket is twice as large as the previous one, in ascending order
+        gsl::index index = static_cast<gsl::index>(ceil(log2(size)));
+        assert((1ull << index) >= size); // This must be true unless there were some strange rounding issues
+
+        // The smallest bucket is 2^n bytes large, where n = c_minResourceSizeExponent
+        index = std::max<gsl::index>(index, c_minResourceSizeExponent);
+        index -= c_minResourceSizeExponent;
+
+        return index;
+    }
+
+    /*static*/ uint64_t DmlReservedResourceSubAllocator::GetBucketSizeFromIndex(gsl::index index)
+    {
+        return (1ull << (index + c_minResourceSizeExponent));
+    }
+
+    void DmlReservedResourceSubAllocator::SetDefaultRoundingMode(AllocatorRoundingMode roundingMode)
+    {
+        // Nothing to do here; kept for compatibility with the bucketized allocator
+    }
+
+    static bool GetTilingEnabled(ID3D12Device* device)
+    {
+        D3D12_FEATURE_DATA_D3D12_OPTIONS options = {};
+        if (SUCCEEDED(device->CheckFeatureSupport(
+                D3D12_FEATURE_D3D12_OPTIONS,
+                &options,
+                sizeof(options))))
+        {
+            return options.TiledResourcesTier >= D3D12_TILED_RESOURCES_TIER_1;
+        }
+
+        return false;
+    }
+
+    static uint64_t GetMaxHeapSizeInTiles()
+    {
+        return DmlReservedResourceSubAllocator::kDefaultMaxHeapSizeInTiles;
+    }
+
+    DmlReservedResourceSubAllocator::DmlReservedResourceSubAllocator(
+        ID3D12Device* device,
+        std::shared_ptr<ExecutionContext> context,
+        ID3D12CommandQueue* queue,
+        const D3D12_HEAP_PROPERTIES& heap_props,
+        D3D12_HEAP_FLAGS heap_flags,
+        D3D12_RESOURCE_FLAGS resource_flags,
+        D3D12_RESOURCE_STATES initial_state)
+        : m_device(device),
+        m_context(context),
+        queue_(queue),
+        heap_properties_(heap_props),
+        heap_flags_(heap_flags),
+        resource_flags_(resource_flags),
+        initial_state_(initial_state),
+        tiling_enabled_(GetTilingEnabled(device)),
+        max_heap_size_in_tiles_(GetMaxHeapSizeInTiles())
+    {
+    }
+
+    absl::optional<DmlHeapAllocation> DmlReservedResourceSubAllocator::TryCreateTiledAllocation(uint64_t size_in_bytes)
+    {
+        DmlHeapAllocation allocation = {};
+
+        // The allocation may be larger than the requested size to ensure a whole
+        // number of tiles.
+        const uint64_t resource_size_in_tiles = 1 + (size_in_bytes - 1) / D3D12_TILED_RESOURCE_TILE_SIZE_IN_BYTES;
```

Comment:
The entire DML EP is 🐪 camelCase. Please use VS's right-click Rename option to for 🐍 `resource_size_in_tiles` -> 🐪 `resourceSizeInTiles`, to keep intra-file consistency.